IrisHub/SpaceBar
# server: address npm ERR! server crash

### Description of Changes
* Attempts to fix the server crash (logs below). 
* The webrtc types package should not affect runtime since it's just used for writing TS and then gets transpiled to JS anyway, but since it's a lightweight package I thought I'd add it back for easier dev. 
```
Oct 03 22:21:21 spacebar-5 heroku/web.1 State changed from up to crashed
Oct 03 22:21:21 spacebar-5 heroku/web.1 State changed from crashed to starting
Oct 03 22:21:21 spacebar-5 app/web.1 Segmentation fault
Oct 03 22:21:21 spacebar-5 app/web.1 npm ERR! code ELIFECYCLE
Oct 03 22:21:21 spacebar-5 app/web.1 npm ERR! errno 139
Oct 03 22:21:21 spacebar-5 app/web.1 npm ERR! server@0.1.0 start: node out/app.js
Oct 03 22:21:21 spacebar-5 app/web.1 npm ERR! Exit status 139
Oct 03 22:21:21 spacebar-5 app/web.1 npm ERR! 
Oct 03 22:21:21 spacebar-5 app/web.1 npm ERR! Failed at the server@0.1.0 start script.
Oct 03 22:21:21 spacebar-5 app/web.1 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
Oct 03 22:21:21 spacebar-5 app/web.1 npm ERR! A complete log of this run can be found in:
Oct 03 22:21:21 spacebar-5 app/web.1 npm ERR!     /app/.npm/_logs/2023-10-04T05_21_21_255Z-debug.log
Oct 03 22:21:21 spacebar-5 heroku/web.1 Process exited with status 139
```

### Testing Procedure
* It happens quite spuriously. To test, I went line by line on the changes introduced by https://github.com/IrisHub/SpaceBar/pull/262 which could have potentially caused this. Three possible hypotheses: 
> 1. Could have been introduced by the `node-pre-gyp` package. As per my understanding, this is used by the wrtc package. Any now that we're installing explicitly (to fix some pinning warnings), there could be conflicts that only get triggered during a particular state at runtime, leading to the crash. If this hypothesis is correct, then removing `node-pre-gyp` should prevent this. 
> 2. There could have been some stale yarn build cache left on the dyno before the upgrade to berry which was not properly cleaned up. To address this, I went into the dyno and purged all artifacts, reinstalled packages, then did a clean deploy. 
> 3. Upgrading packages like node may have introduced incompatibilities with other deps that lead to spurious crashes. Since hypothesis this quite ambiguous, the approach I'm using is that if 1 and 2 don't fix it, then I investigate package upgrades.

* Builds fine locally, and deployed this on Heroku; things seem to work more or less as usual.
